src/routes/safe/components/AddressBook/ExportEntriesModal/index.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { ReactElement, useEffect, useState } from 'react'
+import { format } from 'date-fns'
+import { useSelector, useDispatch } from 'react-redux'
+import { CSVDownloader, jsonToCSV } from 'react-papaparse'
+import styled from 'styled-components'
+
+import { enhanceSnackbarForAction, getNotificationsFromTxType } from 'src/logic/notifications'
+import enqueueSnackbar from 'src/logic/notifications/store/actions/enqueueSnackbar'
+import { TX_NOTIFICATION_TYPES } from 'src/logic/safe/transactions'
+
+import { Button, Loader, Text } from '@gnosis.pm/safe-react-components'
+
+import { addressBookSelector } from 'src/logic/addressBook/store/selectors'
+import { AddressBookState } from 'src/logic/addressBook/model/addressBook'
+
+import { useStyles } from './style'
+
+import { Modal } from 'src/components/Modal'
+import Img from 'src/components/layout/Img'
+import Row from 'src/components/layout/Row'
+
+import SuccessSvg from './assets/success.svg'
+import ErrorSvg from './assets/error.svg'
+import LoadingSvg from './assets/wait.svg'
+
+type ExportEntriesModalProps = {
+  isOpen: boolean
+  onClose: () => void
+}
+
+const BodyImage = styled.div`
+  grid-row: 1;
+`
+export const ExportEntriesModal = ({ isOpen, onClose }: ExportEntriesModalProps): ReactElement => {
+  const classes = useStyles()
+  const dispatch = useDispatch()
+  const addressBook: AddressBookState = useSelector(addressBookSelector)
+  const [loading, setLoading] = useState<boolean>(true)
+  const [error, setError] = useState<string | undefined>('')
+  const [csvData, setCsvData] = useState<string>('')
+  const [doRetry, setDoRetry] = useState<boolean>(false)
+
+  const date = format(new Date(), 'MM-dd-yyyy')
+
+  const handleClose = () =>
+    setTimeout(() => {
+      if (!loading) {
+        const notification = getNotificationsFromTxType(TX_NOTIFICATION_TYPES.ADDRESSBOOK_EXPORT_ENTRIES)
+        const action = error
+          ? notification.afterExecution.afterExecutionError
+          : notification.afterExecution.noMoreConfirmationsNeeded
+        dispatch(enqueueSnackbar(enhanceSnackbarForAction(action)))
+      }
+      onClose()
+    }, 600)
+
+  useEffect(() => {
+    const handleCsvData = () => {
+      if (!isOpen && !doRetry) return
+      setLoading(true)
+      setError('')
+      try {
+        setCsvData(jsonToCSV(addressBook))
+      } catch (e) {
+        setError(e.message)
+      }
+      if (csvData) {
```

Comment:
if this `if` needed? If the `try` was a success you can be sure enough to reset `loading ` an `doRetry`, you will only need to `return` after `setError` in the `catch`.